src/desert/_make.py

```diff
@@ -119,9 +117,9 @@ def class_schema(
         raise desert.exceptions.NotAnAttrsClassOrDataclass(clazz)
 
     # Copy all public fields of the dataclass to the schema
-    attributes = {
-        field.name: field for field in fields if not field.name.startswith("_")
-    }
+    attributes: t.Dict[
+        str, t.Union[dataclasses.Field, attr.Attribute, marshmallow.fields.Field]
```

Comment:
Mmm, like it should be turned inside out?  A union of two dicts such that the keys must always be either `attrs` or `dataclasses`?  More like above where it is...

```python
fields: t.Union[t.Tuple[dataclasses.Field, ...], t.Tuple[attr.Attribute, ...]]
```

or here as...

```python
t.Union[
    t.Dict[str, t.Union[dataclasses.Field, marshmallow.fields.Field]],
    t.Dict[str, t.Union[attr.Attribute, marshmallow.fields.Field]],
]
```

Reply:
I don't know why Marshmallow leaked in there...  Anyways, mypy doesn't act the way I expect.  Digging into it.

https://mypy-play.net/?mypy=latest&python=3.10&gist=0756f026545af8559782e541c5351d80
```python
import typing

these_or_those: typing.Union[typing.List[int], typing.List[str]] = []

reveal_type([x for x in these_or_those])
```

```
main.py:3: error: Incompatible types in assignment (expression has type "List[<nothing>]", variable has type "Union[List[int], List[str]]")
main.py:5: note: Revealed type is "builtins.list[Union[builtins.int*, builtins.str*]]"
Found 1 error in 1 file (checked 1 source file)
```

Reply:
I dunno.  I was directed to https://mail.python.org/archives/list/typing-sig@python.org/thread/GYVM5KEE6URE6PAH7UTK6324M7GWSFQS/ as maybe being related but it seems like one way or another we will not have this "fixed" (if it is in fact a bug).

But looking again I'm not entirely convinced this line is even needed.  We reassign (almost) all of the values below in the for loop (that's probably how `marshmallow.fields.Field` leaked in).  "Almost" because the non-init fields are left untouched.  But does that really make sense putting `attrs`/`dataclass` fields in the schema anyways?  It seems not.  So maybe just deleting this line is the "real" response to mypy complaining.  Let me know what you think.  If this is bad then we should add a test to point it out.